sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1461,7 +1464,8 @@ public boolean apply(@Nonnull String input) {
             }
           }
           convertedOptions.put(entry.getKey(), MAPPER.convertValue(values, type));
-        } else if (SIMPLE_TYPES.contains(returnType) || returnType.isEnum()) {
+        } else if (SIMPLE_TYPES.contains(returnType) || returnType.isEnum()
+          || returnType.equals(ValueProvider.class)) {
```

Comment:
hmm. Not sure why we want to treat VPs like they are strings or enums. Can't I have a `ValueProvider<List<Integer>>`?